gitguard.py

```diff
@@ -249,3 +254,128 @@ def get_commit_history(repo_link, author_name=None, start=None, end=None, path=N
         commit['commit_message'] = commit_history[i]["commit"]["message"]
         history.append(commit)
     return history
+
+def _remove_readonly(func, path, excinfo):
+    """
+    Helper function to remove change read-only file status to editable
+    Source: http://stackoverflow.com/questions/1889597/deleting-directory-in-python
+    """
+    os.chmod(path, stat.S_IWRITE)
+    func(path)
+
+def _clone_repo(repo_link, destination=None):
+    """
+    Helper function to clone a repo to destination. If folder already exists, delete all version
+    
+    Args:
+        repo_link:      owner/repo format
+        destination:    local file path, set to default if not supplied
+
+    Return:
+        reference to the cloned repo
+    """
+    if not destination:
+        destination = "%s%s%s" % ("C:/", repo_link, "-gitguard")
+    if os.path.exists(destination):
+        shutil.rmtree(destination, onerror=_remove_readonly)
+    repo_url = "%s%s%s" % ("https://github.com/", repo_link, ".git");
+    from git import Repo
+    repo = Repo.clone_from(repo_url , destination, branch="master")
```

Comment:
Keep in mind about private repository
I'm not too sure about how gitpython works, but if you use ssh instead of https, it might automatically use the ssh key the person has.
So, I would suggest letting user insert the link themselves. i.e: ```git@github.com:tjonganthony/some_repo.git``` as input instead of ```tjonganthony/some_repo```.

Then, try looking at how to clone using some access privilege to be able to clone private repo.
If it's too bothersome, do include in the documentation that we can only do analysis on public repo.